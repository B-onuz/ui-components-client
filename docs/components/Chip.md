# Chip (Component)

Componente chip usado para listar status.

## Como utilizar

```js
import Chip from 'ui-components/lib/components/Chip'
```

```jsx
<Chip color={'primary'} text={'status'} />
```

## Variações

### Variant

```jsx
<Chip color={'secondary'} />
```

Escala no tema: `theme.variantScale`

## Styled-system props

- space https://styled-system.com/table#space
- layout https://styled-system.com/table#layout
