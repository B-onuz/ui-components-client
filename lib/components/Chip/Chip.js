"use strict";

exports.__esModule = true;
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _styledSystem = require("styled-system");

var _variants = require("./variants");

var _this = void 0,
    _jsxFileName = "/Users/master/projects/bonuz/ui-components/src/components/Chip/Chip.js";

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n  ", "\n  background: ", ";\n  padding: 2px 10px 4px;\n  border-radius: 25px;\n  white-space: nowrap;\n  overflow: hidden;\n  text-overflow: ellipsis;\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  ", "\n  display: flex;\n  align-items: center;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

var StyledChip = _styledComponents["default"].div(_templateObject(), _styledSystem.space);

var StyledText = _styledComponents["default"].span(_templateObject2(), _variants.color, function (_ref) {
  var theme = _ref.theme,
      color = _ref.color;
  return "" + theme.colors[color];
});

var Chip = function Chip(_ref2) {
  var color = _ref2.color,
      text = _ref2.text,
      rest = _objectWithoutPropertiesLoose(_ref2, ["color", "text"]);

  return /*#__PURE__*/_react["default"].createElement(StyledChip, _extends({}, rest, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 24,
      columnNumber: 5
    }
  }), /*#__PURE__*/_react["default"].createElement(StyledText, {
    color: color,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 25,
      columnNumber: 7
    }
  }, text));
};

var _default = Chip;
exports["default"] = _default;
module.exports = exports.default;