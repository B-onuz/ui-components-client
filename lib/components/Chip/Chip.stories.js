'use strict'

exports.__esModule = true
exports.ChipExample = exports['default'] = void 0

var _react = _interopRequireDefault(require('react'))

var _Chip = _interopRequireDefault(require('./Chip'))

var _reflexbox = require('reflexbox')

var _styledSystem = require('styled-system')

var _this = void 0,
  _jsxFileName = '/home/junior/Fontes/ui-components-client/src/components/Chip/Chip.stories.js'

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : { default: obj }
}

var _default = {
  title: 'Chip',
  component: _Chip['default'],
}
exports['default'] = _default

var ChipExample = function ChipExample() {
  return /*#__PURE__*/ _react['default'].createElement(
    _reflexbox.Flex,
    {
      m: 3,
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 12,
        columnNumber: 3,
      },
    },
    /*#__PURE__*/ _react['default'].createElement(_Chip['default'], {
      mr: 2,
      color: 'primary',
      text: 'status',
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 13,
        columnNumber: 5,
      },
    }),
    /*#__PURE__*/ _react['default'].createElement(_Chip['default'], {
      mr: 2,
      color: 'warning',
      text: 'status',
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 14,
        columnNumber: 5,
      },
    }),
    /*#__PURE__*/ _react['default'].createElement(_Chip['default'], {
      mr: 2,
      color: 'info',
      text: 'status',
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 15,
        columnNumber: 5,
      },
    }),
    /*#__PURE__*/ _react['default'].createElement(_Chip['default'], {
      mr: 2,
      color: 'success',
      text: 'status',
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 16,
        columnNumber: 5,
      },
    }),
    /*#__PURE__*/ _react['default'].createElement(_Chip['default'], {
      mr: 2,
      color: 'orange',
      text: 'status',
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 17,
        columnNumber: 5,
      },
    }),
    /*#__PURE__*/ _react['default'].createElement(_Chip['default'], {
      mr: 2,
      color: 'yellow',
      text: 'status',
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 18,
        columnNumber: 5,
      },
    }),
    /*#__PURE__*/ _react['default'].createElement(_Chip['default'], {
      mr: 2,
      color: 'secondary',
      text: 'status',
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 19,
        columnNumber: 5,
      },
    }),
    /*#__PURE__*/ _react['default'].createElement(_Chip['default'], {
      mr: 2,
      color: 'fullDarkRed',
      text: 'status',
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 20,
        columnNumber: 5,
      },
    }),
    /*#__PURE__*/ _react['default'].createElement(_Chip['default'], {
      mr: 2,
      color: 'lightRed',
      text: 'status',
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 21,
        columnNumber: 5,
      },
    }),
    /*#__PURE__*/ _react['default'].createElement(_Chip['default'], {
      mr: 2,
      color: 'olive',
      text: 'status',
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 22,
        columnNumber: 5,
      },
    }),
    /*#__PURE__*/ _react['default'].createElement(_Chip['default'], {
      mr: 2,
      color: 'blue',
      text: 'status',
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 23,
        columnNumber: 5,
      },
    }),
    /*#__PURE__*/ _react['default'].createElement(_Chip['default'], {
      mr: 2,
      color: 'teal',
      text: 'status',
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 24,
        columnNumber: 5,
      },
    })
  )
}

exports.ChipExample = ChipExample
