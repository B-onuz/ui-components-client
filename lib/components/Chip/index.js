"use strict";

exports.__esModule = true;
exports["default"] = void 0;

var _Chip = _interopRequireDefault(require("./Chip"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var _default = _Chip["default"];
exports["default"] = _default;
module.exports = exports.default;