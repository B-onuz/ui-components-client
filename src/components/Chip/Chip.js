import React from 'react'
import styled from 'styled-components'
import { space } from 'styled-system'
import { color } from './variants'

const StyledChip = styled.div`
  ${space}
  display: flex;
  align-items: center;
`

const StyledText = styled.span`
  ${color}
  background: ${({ theme, color }) => `${theme.colors[color]}`};
  padding: 2px 10px 4px;
  border-radius: 25px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
`

const Chip = ({ color, text, ...rest }) => {
  return (
    <StyledChip {...rest}>
      <StyledText color={color}>{text}</StyledText>
    </StyledChip>
  )
}

export default Chip
