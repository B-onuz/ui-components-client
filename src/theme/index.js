const colors = {
  red: '#db2828',
  lightRed: '#ef4d5a',
  mediumRed: '#FF4F5D',
  darkRed: '#9c2543',
  fullDarkRed: '#550e33',
  orange: '#f2711c',
  yellow: '#fbbd08',
  olive: '#b5cc18',
  green: '#21BA45',
  teal: '#00b5ad',
  blue: '#2185d0',
  facebookBlue: '#3B5998',
  blueLink: '#4285F4',
  violet: '#6435c9',
  purple: '#a333c8',
  pink: '#e03997',
  brown: '#a5673f',
  grey: '#767676',
  textGrey: '#8D8D8D',
  lightGrey: '#6D6D6D',
  mediumGrey: '#838383',
  darkGrey: '#9A9A9A',
  boldGray: '#55555A',
  black: '#1b1c1d',
  white: '#fdfdfd',
  warning: '#7A4D05',
  success: '#1E561F',
  info: '#0E566C',
  danger: '#973937',
  cyan: '#1CB5AD',
  disabledGrey: '#ffffff1f',
  disabledGreyText: '#ffffff4d',
}

colors.primary = colors.lightRed
colors.secondary = colors.fullDarkRed
colors.default = '#e0e1e2'

const breakpoints = ['40em', '52em', '64em', '80em']

breakpoints.sm = breakpoints[0]
breakpoints.md = breakpoints[1]
breakpoints.lg = breakpoints[2]
breakpoints.xl = breakpoints[3]

const fontSizes = [12, 14, 16, 20, 24, 32]

export default {
  fontFamily: `'Lato', sans-serif`,
  colors,
  fontSizes,
  breakpoints,
}
