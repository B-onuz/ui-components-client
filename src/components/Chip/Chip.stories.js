import React from 'react'
import Chip from './Chip'
import { Flex, Box } from 'reflexbox'
import { space } from 'styled-system'

export default {
  title: 'Chip',
  component: Chip,
}

export const ChipExample = () => (
  <Flex m={3}>
    <Chip mr={2} color={'primary'} text={'status'} />
    <Chip mr={2} color={'warning'} text={'status'} />
    <Chip mr={2} color={'info'} text={'status'} />
    <Chip mr={2} color={'success'} text={'status'} />
    <Chip mr={2} color={'orange'} text={'status'} />
    <Chip mr={2} color={'yellow'} text={'status'} />
    <Chip mr={2} color={'secondary'} text={'status'} />
    <Chip mr={2} color={'fullDarkRed'} text={'status'} />
    <Chip mr={2} color={'lightRed'} text={'status'} />
    <Chip mr={2} color={'olive'} text={'status'} />
    <Chip mr={2} color={'blue'} text={'status'} />
    <Chip mr={2} color={'teal'} text={'status'} />
  </Flex>
)
