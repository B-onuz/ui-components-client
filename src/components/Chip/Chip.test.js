import React from 'react'
import Chip from './Chip.js'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import theme from '../../theme'
import 'jest-styled-components'

test('Should render Chip', () => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <Chip color={'primary'} />
    </ThemeProvider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
