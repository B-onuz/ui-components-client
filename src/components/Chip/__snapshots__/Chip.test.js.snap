// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Chip 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  color: #fff;
  background: #ef4d5a;
  padding: 2px 10px 4px;
  border-radius: 25px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

<div
  className="c0"
>
  <span
    className="c1"
    color="primary"
  />
</div>
`;
